lib/AI-Agent.xml

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsightsAgent>
+    <AgentLogger>INFO</AgentLogger>
+    <Instrumentation>
+        <BuiltIn enabled="true">
+            <HTTP enabled="true"/>
+        </BuiltIn>
+        <Class name="uk.gov.hmcts.ccd.data.casedetails.CaseAuditEventRepository">
+            <Method name="set"/>
+            <Method name="findByCase"/>
+            <Method name="getCreateEvent"/>
+            <Method name="findByEventId"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.definition.DefaultCaseDefinitionRepository">
+            <Method name="getCaseTypesForJurisdiction"/>
+            <Method name="getCaseType"/>
+            <Method name="getBaseTypes"/>
+            <Method name="getUserRoleClassifications"/>
+            <Method name="getClassificationsForUserRoleList"/>
+            <Method name="getLatestVersion"/>
+            <Method name="getJurisdictions"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.casedetails.DefaultCaseDetailsRepository">
+            <Method name="set"/>
+            <Method name="findById"/>
+            <Method name="findByReference"/>
+            <Method name="lockCase"/>
+            <Method name="findUniqueCase"/>
+            <Method name="findByMetaDataAndFieldData"/>
+            <Method name="getPaginatedSearchMetadata"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.caseaccess.CaseUserAuditRepository">
+            <Method name="auditGrant"/>
+            <Method name="auditRevoke"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.caseaccess.CaseUserRepository">
+            <Method name="grantAccess"/>
+            <Method name="revokeAccess"/>
+            <Method name="findCasesUserIdHasAccessTo"/>
+            <Method name="findCaseRoles"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.caseaccess.DefaultCaseRoleRepository">
+            <Method name="getCaseRoles"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.user.DefaultUserRepository">
+            <Method name="getUserDetails"/>
+            <Method name="getUserDefaultSettings"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.definition.UIDefinitionRepository">
```

Comment:
@timja some of our repositories, for example UIDefinitionRepository, do not interact with the database. They instead issue http calls to external services. Shall we include them as well or this file if purely for database layer Repositories?

Reply:
no http should be automatically instrumented, but needs to be verified. I'll do that soon unless you beat me to it